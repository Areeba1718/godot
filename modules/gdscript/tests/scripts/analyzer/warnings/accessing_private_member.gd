class A:
	@warning_ignore("unused_private_class_variable")
	var __a
	static var __b

	func __c():
		pass

	static func __d():
		pass

class B:
	var a = A.new()
	var b = a.__a
	@warning_ignore("accessing_private_member")
	var c = a.__a
	@warning_ignore("accessing_private_member") var d = a.__a
	var e = a.__a

	var f = A.__b
	@warning_ignore("accessing_private_member")
	var g = A.__b
	@warning_ignore("accessing_private_member") var h = A.__b
	var i = A.__b

	func __init():
		a.__c()
		@warning_ignore("calling_private_method")
		a.__c()
		@warning_ignore("calling_private_method") a.__c()
		a.__c()

		A.__d()
		@warning_ignore("calling_private_method")
		A.__d()
		@warning_ignore("calling_private_method") A.__d()
		A.__d()

	var j = a.__c()
	@warning_ignore("calling_private_method")
	var k = a.__c()
	@warning_ignore("calling_private_method") var l = a.__c()
	var m = a.__c()

	var n = A.__d()
	@warning_ignore("calling_private_method")
	var o = A.__d()
	@warning_ignore("calling_private_method") var p = A.__d()
	var q = A.__d()

func test():
	pass
