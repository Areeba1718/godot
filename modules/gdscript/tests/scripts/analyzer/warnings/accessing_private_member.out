GDTEST_OK
>> WARNING
>> Line: 14
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an external class.
>> WARNING
>> Line: 18
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__a" in an external class.
>> WARNING
>> Line: 20
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an external class.
>> WARNING
>> Line: 24
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "__b" in an external class.
>> WARNING
>> Line: 27
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an external class.
>> WARNING
>> Line: 31
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an external class.
>> WARNING
>> Line: 33
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an external class.
>> WARNING
>> Line: 37
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an external class.
>> WARNING
>> Line: 39
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an external class.
>> WARNING
>> Line: 43
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__c()" in an external class.
>> WARNING
>> Line: 45
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an external class.
>> WARNING
>> Line: 49
>> CALLING_PRIVATE_METHOD
>> Trying calling a private method "__d()" in an external class.
